src/cap32.cpp

```diff
@@ -2091,7 +2100,13 @@ int cap32_main (int argc, char **argv)
 
             case SDL_KEYUP:
                {
-                  dword cpc_key = CPC.InputMapper->CPCkeyFromKeysym(event.key.keysym);
+                  bool apply_shift = false;
+                  if (!shift_down) {
+                     auto idx = static_cast<int>(event.key.keysym.scancode);
+                     apply_shift = shifted_keys[idx];
```

Comment:
IIUC, this is never set back to false if you release the key before releasing shift

Reply:
Yes, this solves the initial bug, but introduces the opposite bug where the key down -> shift down -> key up results in repeating characters. I am closing this PR.